pkg/cmd/server/server.go

```diff
@@ -819,7 +808,13 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 			s.logger.Infof("Disabling controller: %s", controllerName)
 			delete(enabledControllers, controllerName)
 		} else {
-			s.logger.Fatalf("Invalid value for --disable-controllers flag provided: %s. Valid values are: %s", controllerName, strings.Join(disableControllerList, ","))
+			// maybe it is a runtime type controllers, so attempt to remove that
+			if _, ok := enabledRuntimeControllers[controllerName]; ok {
+				s.logger.Infof("Disabling controller: %s", controllerName)
+				delete(enabledRuntimeControllers, controllerName)
+			} else {
+				s.logger.Fatalf("Invalid value for --disable-controllers flag provided: %s. Valid values are: %s", controllerName, strings.Join(disableControllerList, ","))
```

Comment:
Is this an error for which we want to halt execution?

Also, this happens only if `controllerName ` is not in `enabledControllers`.
Suggest making a separate function for the `for` loop from Line 806 so that we can have unit tests for this.

Reply:
> Also, this happens only if controllerName is not in enabledControllers.

That is correct, but I don't understand if you are pointing out an issue.

> Is this an error for which we want to halt execution?
It is the original behavior: https://github.com/vmware-tanzu/velero/blob/976484553062262324cc1304607ea00841262db8/pkg/cmd/server/server.go#L822. We can re-evaluate, but not sure this PR is the place to do it. Added this question to today's community meeting.

Reply:
Logged an issue for this: https://github.com/vmware-tanzu/velero/issues/2889

Reply:
The other point I was trying to make is that we are halting executions only when the`if` on line 812 is `false` and there is a misconfiguration in the `enabledRuntimeControllers`.

We should be halting execution when there is a misconfiguration in either "kinds" of controllers and not just one.

Reply:
>  only when the if on line 812 is false 

I don't think so.. It fails if either ln 807 is false OR ln 812 if false.

If there is a match with the passed in config, we can move on and check the next input. We only care to keep checking the second list if a passed in input wasn't found in the first list. 

Am I getting this wrong?

Reply:
Lines 811-817 are in the `else` block if the `if` at line 807.
So line 816, which has the `s.logger.Fatalf` will be executed only if the condition on line 807 is `false`

Reply:
Right. Why would it ever execute if the condition on line 807 is true?

Reply:
that's the point. We want to halt execution if there is any misconfiguration in the enabled/disabled controllers, not just for "runtime" controllers

Reply:
I see it now!